Check for 64bit FP instructions availability in libm before using them.

Index: newlib/libm/machine/arm/s_rint.c
--- newlib/libm/machine/arm/s_rint.c.orig
+++ newlib/libm/machine/arm/s_rint.c
@@ -24,7 +24,7 @@
    NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. */
 
-#if __ARM_ARCH >= 8 && !defined (__SOFTFP__)
+#if __ARM_ARCH >= 8 && (__ARM_FP & 0x8) && !defined (__SOFTFP__)
 #include <math.h>
 
 double
